README.md

```diff
@@ -1,2 +1,208 @@
+* __GOCMDDIR__ - The location of the directory containing the sub-directories for the executables to be built.  Defaults to `./cmd`
+* __GOROOTTARGET__ - Can be set to the name of a single target from `GOTARGETS` if the main executable code for it is in the top level directory.  No default
+* __GOSRC__ - A list of the Go source files.  Defaults to all Go source files recursively found in the current directory
+* __GO__ - The Go executable to run.  Defaults to `go`
+* __GOFLAGS__ - Flags to pass to `go build`.  No default
+* __GOTESTTARGET__ - The target to pass to `go test`.  Defaults to `./...`, which means all tests in the project
+* __GOTESTFLAGS__ - Flags to pass to `go test`.  Defaults to `-v -race`
+
+
+## Copy
+
+Recipes for copying files as part of the build are included in `copy-common.mk`
+
+### Targets
+
+The primary targets provided for Go are:
+
+* __build__ - Copies files into the build directory
+* __clean__ - Removes build targets
+
+Each primary target depends on three sub-targets, which use the prefixes `pre-`,
+`standard-`, and `post-`.  E.g., the sub-targets for the `build` target are
+`pre-build`, `standard-build`, and `post-build`.  The common recipes only define
+actions for `standard-` sub-target.  The others are provided to allow custom actions.
+
+### Variables
+
+These variables are used by the standard targets and may be overridden:
+
+* __PROJECT__ - The name of the project being built.  Defaults to the name of the current directory (which should be also the name of the repository)
+* __BUILDDIR__ - The location that build artifacts will be placed in.  Defaults to `./build`
+* __COPY__ - The program to use to copy files.  Defaults to `cp`
+* __COPYTARGETS__ - A list of files and directories to be copied into `BUILDDIR`.  No default
+* __COPYFLAGS__ - Flags to pass to `COPY`.  Defaults to `-r`
+
+## Using
+
+The normal way to use the common make recipes is to include the appropriate files in
+the Makefile for a repository.  For a simple Go microservice, just including
+`go-common.mk` may be all that is required in the Makefile, for example:
+
+```make
+include go-common.mk
+```
+
+### Include
+
+In all cases the Makefile will ultimately reference one or more of the `*-common.mk`
+files in this package, usually through the use of the `include` directive.  However,
+there are a number of options on how this directive can be resolved.
+
+#### Dynamic download
+
+In this option, the Makefile includes a target for creating the common make file that
+is included by downloading it from Github as well as an `include` directive.  In this
+case the name of the common make file that is downloaded should be added to
+`.gitignore` for the package so that it is not inadvertently committed.
+
+`curl` will be used in the following examples to illustrate a variety of forms this
+could take.
+
+Get the most recent version of `go-common-mk`:
+
+```make
+include go-common.mk
+
+go-common.mk:
```

Comment:
Will this automatically be executed by the above include?

Reply:
> Will this automatically be executed by the above include?

No, you would need to manually add this target